presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunner.java

```
@@ -0,0 +1,301 @@
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<QualifiedObjectName> listTables(Session session, String catalog, String schema)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean tableExists(Session session, String table)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void installPlugin(Plugin plugin)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createCatalog(String catalogName, String connectorName, Map<String, String> properties)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void loadFunctionNamespaceManager(String functionNamespaceManagerName, String catalogName, Map<String, String> properties)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Lock getExclusiveLock()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getNodeCount()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Session getDefaultSession()
+    {
+        return null;
+    }
+
+    @Override
+    public MaterializedResult execute(Session session, String sql)
+    {
+        String[] command = {
+                "/opt/presto-cli",
+                "--server",
+                "presto-coordinator:" + coordinatorPort,
+                "--catalog",
+                catalog,
+                "--schema",
+                schema,
+                "--execute",
+                sql
+        };
+
+        Container.ExecResult execResult = null;
+        try {
+            execResult = coordinator.execInContainer(command);
+        }
+        catch (IOException e) {
+            fail("Presto CLI failed with error message: " + e.getMessage());
```

It's clearer to the end user to let the exception bubble up and fail the test than convert it into a fail call.

---

```
@Override
   public MaterializedResult execute(Session session, String sql) { ... }
```
Can we add exceptions to an override method? If so, aren't we required to edit its base class and the other children of the parent class?

---

I think this will also introduce many changes in `com/facebook/presto/sql/query/QueryAssertions.java` wherever the execute gets called.